### ISSUE TITLE
Flaky-test: TestPulsarAuth.testPulsarSqlAuth hangs and makes the test run to get cancelled by a timeout

### ISSUE THREAD

**Issue by lhotari** [apache/pulsar#19032]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Example failure

https://github.com/apache/pulsar/actions/runs/3759470154/jobs/6390308187#step:11:481

### Exception stacktrace

```
"main" #1 prio=5 os_prio=0 cpu=19741.86ms elapsed=29.78s tid=0x00007f1fa8028e40 nid=0xb9bd waiting on condition  [0x00007f1fad2b0000]
   java.lang.Thread.State: WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@17.0.5/Native Method)
	- parking to wait for  <0x0000100027c0f240> (a java.util.concurrent.CompletableFuture$Signaller)
	at java.util.concurrent.locks.LockSupport.park(java.base@17.0.5/LockSupport.java:211)
	at java.util.concurrent.CompletableFuture$Signaller.block(java.base@17.0.5/CompletableFuture.java:1864)
	at java.util.concurrent.ForkJoinPool.unmanagedBlock(java.base@17.0.5/ForkJoinPool.java:3463)
	at java.util.concurrent.ForkJoinPool.managedBlock(java.base@17.0.5/ForkJoinPool.java:3434)
	at java.util.concurrent.CompletableFuture.waitingGet(java.base@17.0.5/CompletableFuture.java:1898)
	at java.util.concurrent.CompletableFuture.get(java.base@17.0.5/CompletableFuture.java:2072)
	at org.apache.pulsar.client.impl.PulsarClientImpl.close(PulsarClientImpl.java:715)
	at org.apache.pulsar.sql.presto.PulsarAuth.checkTopicAuth(PulsarAuth.java:102)
	at org.apache.pulsar.sql.presto.TestPulsarAuth.testPulsarSqlAuth(TestPulsarAuth.java:187)
```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by lhotari** [comment]
might be related to #13964

**Comment by tisonkun** [comment]
https://github.com/apache/pulsar/issues/13964 has been fixed. Closing as no more report for this case. Feel free to open a new ticket or reopen this one if it comes back.
